core/src/plugins/filed/python/pyfiles/BareosFdPluginLocalFileset.py

```diff
@@ -132,3 +132,74 @@ def start_backup_job(self):
             return bareosfd.bRC_Error
         else:
             return bareosfd.bRC_OK
+
+
+
+    def plugin_io_open(self, IOP):
+        self.FNAME = IOP.fname
+        bareosfd.DebugMessage(250, "io_open: self.FNAME is set to %s\n" % (self.FNAME))
+        if os.path.isdir(self.FNAME):
+            bareosfd.DebugMessage(100, "%s is a directory\n" % (self.FNAME))
+            self.fileType = "FT_DIR"
+            bareosfd.DebugMessage(
+                100,
+                "Did not open file %s of type %s\n" % (self.FNAME, self.fileType),
+            )
+            return bareosfd.bRC_OK
+        elif os.path.islink(self.FNAME):
+            self.fileType = "FT_LNK"
+            bareosfd.DebugMessage(
+                100,
+                "Did not open file %s of type %s\n" % (self.FNAME, self.fileType),
+            )
+            return bareosfd.bRC_OK
+        elif os.path.exists(self.FNAME) and stat.S_ISFIFO(os.stat(self.FNAME).st_mode):
+            self.fileType = "FT_FIFO"
+            bareosfd.DebugMessage(
+                100,
+                "Did not open file %s of type %s\n" % (self.FNAME, self.fileType),
+            )
+            return bareosfd.bRC_OK
+        else:
+            self.fileType = "FT_REG"
+            bareosfd.DebugMessage(
+                150,
+                "file %s has type %s - trying to open it\n"
+                % (self.FNAME, self.fileType),
+            )
+        try:
+            if IOP.flags & (os.O_CREAT | os.O_WRONLY):
+                bareosfd.DebugMessage(
+                    100,
+                    "Open file %s for writing with %s\n" % (self.FNAME, IOP),
+                )
+                dirname = os.path.dirname(self.FNAME)
+                if not os.path.exists(dirname):
+                    bareosfd.DebugMessage(
+                        100,
+                        "Directory %s does not exist, creating it now\n" % (dirname),
+                    )
+                    os.makedirs(dirname)
+                self.file = open(self.FNAME, "wb")
+            else:
+                bareosfd.DebugMessage(
+                    100,
+                    "Open file %s for reading with %s\n" % (self.FNAME, IOP),
+                )
+                self.file = open(self.FNAME, "rb")
+
+            # do io in core
+            #IOP.filedes = self.file.fileno()
+            #IOP.status =  bareosfd.io_status_core
+            ##IOP.status = 1
+
+            #  do io in plugin
+            IOP.status = bareosfd.io_status_plugin
+            ##IOP.status = 0
```

Comment:
test code should be cleaned up before merge

Reply:
IO in core or IO via plugin can now be selected via plugin options.